src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -685,71 +685,94 @@ class AbstractUtxoCoin extends BaseCoin {
     const txb = bitcoin.TransactionBuilder.fromTransaction(transaction, this.network);
     this.prepareTransactionBuilder(txb);
 
+    const getUnspentInfo = function(self, txPrebuild, index) {
```

Comment:
use arrow syntax (`const getUnspentInfo = (txPrebuild, index) => `) so that you don't have to pass `this` as `self`

In cases where you can't use arrow syntax  (e.g. `co`-wrapped generators)  and have to use `co(function *(){}.bind(this));` instead

Reply:
this function here has a less than elegant interface

```
const unspentInfo = getUnspentInfo(...);
unspentInfo.unspentInfo // moar info
unspentInfo.unspentInfo.unspent // this also contains unspent info!
```

It looks like you added this to have some code reuse and also have values ready for `signatureIssues`

I suggest we call this `getVerificationContext()` instead and move `isP2wsh` and `isBitGoTaintedUnspent` next to `inputIndex` and `unspent` to flatten the thing a bit

Reply:
since you also use it during signing maybe `getSignatureContext()`

Reply:
Done.